### DISCUSSION TITLE
'NoneType' object has no attribute 'refresh_from_task'  Error when manually running task instance

### DISCUSSION THREAD

**Discussion by a-pertsev** [apache/airflow#19366] [General]
### Apache Airflow version

2.2.1 (latest released)

### Operating System

Ubuntu 20.04 LTS

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==2.3.0
apache-airflow-providers-apache-cassandra==2.1.0
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-google==6.0.0
apache-airflow-providers-http==2.0.1
apache-airflow-providers-imap==2.0.1
apache-airflow-providers-jdbc==2.0.1
apache-airflow-providers-mysql==2.1.1
apache-airflow-providers-postgres==2.3.0
apache-airflow-providers-presto==2.0.1
apache-airflow-providers-slack==4.1.0
apache-airflow-providers-sqlite==2.0.1

### Deployment

Virtualenv installation

### What happened

Got error when manually run newly created task in already finished dag run (in UI).

### What you expected to happen

"Run" button should work without exceptions.

### How to reproduce

1. Define a dag with some tasks.
2. Create dag run (manually or by schedule)
3. Add new task into dag, deploy code
4. Select new task in UI and press "Run" button

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
You should provide more reproducible examples (dags that you use), logs detailed information about your deployment. - otherwise it is impossible to act on this. I will convert it to discussion now until this information is provided - then we can try to asses if the eror is real or maybe there is somethign wrong with your deployment.

**Reply by a-pertsev** [reply]
here is some code examples:
```python
from datetime import datetime

from airflow.operators.python_operator import PythonOperator
from airflow import DAG

with DAG('a_dag',  schedule_interval=None, default_args={'start_date': datetime(2018, 1, 1, 0, 0, 0)}) as dag:
    def success(*args, **kwargs):
        print('ok')

    some_task = PythonOperator(task_id='some_tasks', python_callable=success)
```
and with new task

```python
from datetime import datetime

from airflow.operators.python_operator import PythonOperator
from airflow import DAG

with DAG('a_dag',  schedule_interval=None, default_args={'start_date': datetime(2018, 1, 1, 0, 0, 0)}) as dag:
    def success(*args, **kwargs):
        print('ok')

    some_task = PythonOperator(task_id='some_tasks', python_callable=success)
    new_task = PythonOperator(task_id='new_task', python_callable=success)
```

and steps to reproduce:

1. Define "a_dag" with "some_tasks".
2. Create dag run (manually or by schedule)
3. Add "new_task" into dag, deploy code
4. Select "new_task" in UI and press "Run" button

**Reply by potiuk** [reply]
Yep. Reproduced it.

**Comment by potiuk** [comment]
Created an issue back #19477
